### PR TITLE
Add --node-taints to az aks command

### PR DESCRIPTION
**Pull request by Xinyue-Wang**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks create --node-taints "taints": create cluster with all nodepool have "taints"
az aks update --node-taints "taints": update all nodepool  of give cluster to have "taints"

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -4595,6 +4595,56 @@ def test_aks_update_label_msi(self, resource_group, resource_group_location):
             self.check('agentPoolProfiles[0].nodeLabels.label2', None),
         ])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_update_taint_msi(self, resource_group, resource_group_location):
```
[the remark this answers is not in the document]
Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:4600]: > cli.azure.cli.core.azclierror: Traceback (most recent call last):
>   File "/home/fumingzhang/.virtualenvs/nEnv/lib/python3.10/site-packages/knack/invocation.py", line 113, in _validation
>     self._validate_arg_level(parsed_ns)
>   File "/home/fumingzhang/azure-cli/src/azure-cli-core/azure/cli/core/commands/__init__.py", line 859, in _validate_arg_level
>     validator(**self._build_kwargs(validator, ns))
>   File "/home/fumingzhang/tmp/azure-cli-extensions/src/aks-preview/azext_aks_preview/_validators.py", line 229, in validate_taints
>     if namespace.node_taints is not None and namespace.node_taints != '':
> AttributeError: 'Namespace' object has no attribute 'node_taints'

The parameter name of the newly added option (to command aks create/update) is `nodepool_taints`, which is different from the existing one `node_taints` in commands aks nodepool add/update. Please add a new validator function.

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -670,6 +670,7 @@ def aks_create(
     node_count=3,
     nodepool_tags=None,
     nodepool_labels=None,
+    nodepool_taints=None,
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:673]: It seems the value is not set to the model, don't we need some kind set_up or update function? Or maybe you could refer to another option `--nodepool-labels`, https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py#L821-L824

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 3]
```diff
@@ -2945,6 +2945,7 @@ def get_special_parameter_default_value_pairs_list(self) -> List[Tuple[Any, Any]
             (self.context.get_cluster_autoscaler_profile(), None),
             (self.context.get_api_server_authorized_ip_ranges(), None),
             (self.context.get_nodepool_labels(), None),
+            (self.context.get_nodepool_taints(), None),
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2948]: You'll need to add a getter function to the mc decorator context to get the value from agent pool decorator context.

See https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py#L778

File: src/aks-preview/azext_aks_preview/_validators.py [hunk 4]
```diff
@@ -228,6 +228,24 @@ def validate_load_balancer_outbound_ip_prefixes(namespace):
             raise CLIError(
                 "--load-balancer-outbound-ip-prefixes cannot contain whitespace")
 
+def validate_nodepool_taints(namespace):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_validators.py:231]: Duplicate with the code in getter?

**Review by FumingZhang — COMMENTED** [review]
Please resolve conflicts.

File: src/aks-preview/azext_aks_preview/_params.py [hunk 5]
```diff
@@ -146,7 +147,7 @@
     validate_ssh_key_for_update,
     validate_start_date,
     validate_start_time,
-    validate_taints,
+    validate_nodepool_taints,
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:150]: duplicated imports with line 132

**Review by FumingZhang — APPROVED** [review]
LGTM

File: src/aks-preview/azext_aks_preview/_params.py [hunk 6]
```diff
@@ -376,6 +377,7 @@ def load_arguments(self, _):
                    help='space-separated tags: key[=value] [key[=value] ...]. Use "" to clear existing tags.')
         c.argument('nodepool_labels', nargs='*', validator=validate_nodepool_labels,
                    help='space-separated labels: key[=value] [key[=value] ...]. See https://aka.ms/node-labels for syntax of labels.')
+        c.argument('nodepool_taints', type=str, validator=validate_nodepool_taints)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:380]: By default this is with type string, you could omit `type=str` here

File: src/aks-preview/azext_aks_preview/_params.py [hunk 7]
```diff
@@ -519,6 +521,7 @@ def load_arguments(self, _):
         c.argument('max_count', type=int, validator=validate_nodes_count)
         c.argument('nodepool_labels', nargs='*', validator=validate_nodepool_labels,
                    help='space-separated labels: key[=value] [key[=value] ...]. See https://aka.ms/node-labels for syntax of labels.')
+        c.argument('nodepool_taints', type=str, validator=validate_nodepool_taints)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:524]: same here

File: src/aks-preview/azext_aks_preview/_validators.py [hunk 8]
```diff
@@ -229,19 +229,32 @@ def validate_load_balancer_outbound_ip_prefixes(namespace):
...
-            if taint == "":
-                continue
-            found = regex.findall(taint)
-            if not found:
-                raise CLIError('Invalid node taint: %s' % taint)
+    if taint == "":
+        return
+    found = regex.findall(taint)
+    if not found:
+        raise CLIError('Invalid node taint: %s' % taint)
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/_validators.py:257]:
```suggestion
        raise ArgumentUsageError('Invalid node taint: %s' % taint)
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**